chingu-x/chingu-connect
# New/#67 react router

## Purpose
* Implement react-router for frontend pages
* Connect FE with BE api

## Changes/Additions
* Implemented react-router and container to house app
* Created a GitHub auth button for logging in
* Created a navbar with `Help Board`, `Profile`, and `Sign Out` links
* Created a landing page component
* Created a Profile page component
* Created a Help Board (connections feed) page component
* Created a Not Found component to catch all error routes
* Utilizes Apollo graphQL for querying 
* Utilizes axios for BE call requests

## Notes
* I tried commenting and documenting as best as I could
* Worked with @Bigghead to create a hack-y method for calling to our BE just for development purposes (created a `/user` route on the BE in `app.js`
* Hard-coded routes on the frontend for development purposes
* Using state in `App.js` to store user info. I think there's a way to do it with the client Apollo but unsure at the moment how

## Testing
* Download this PR to local device
* Grab new dependencies
* Try logging in on the frontend
* Test all routes
  * Profile route should show avatar and username

**UPDATE**: @Bigghead tested everything
